projectroles/forms.py

```diff
@@ -338,20 +338,48 @@ def _set_app_setting_widget(self, app_name, s_field, s_key, s_val):
                 )
             else:
                 json_data = self.app_settings.get_default(
-                    app_name=app_name, setting_name=s_key
+                    app_name=app_name,
+                    setting_name=s_key,
+                    project=None,
                 )
             self.initial[s_field] = json.dumps(json_data)
         else:
             if s_val.get('options'):
-                self.fields[s_field] = forms.ChoiceField(
-                    choices=[
-                        (int(option), int(option))
-                        if s_val['type'] == 'INTEGER'
-                        else (option, option)
-                        for option in s_val.get('options')
-                    ],
-                    **setting_kwargs
-                )
+                if callable(s_val.get('default')):
+                    if self.instance.pk:
+                        self.initial[s_field] = s_val.get('default')(
+                            project=self.instance
+                        )
+                        self.fields[s_field] = forms.ChoiceField(
+                            choices=[
+                                (str(option), str(option))
+                                for option in s_val.get('options')(
+                                    project=self.instance
+                                )
+                            ],
+                            **setting_kwargs
+                        )
+                    else:
+                        self.initial[s_field] = s_val.get('default')(
+                            project=None
+                        )
+                        self.fields[s_field] = forms.ChoiceField(
+                            choices=[
```

Comment:
Something still seems to be going wrong here, we get an option where both the value and label are "Project UUID".

![2023-03_form_option_bug2](https://user-images.githubusercontent.com/11211145/224261902-6a012660-5b34-409f-aefb-e7f0741bafa1.png)

Reply:
To be honest, I still do not understand what is designed to be here( Befire there were implemented tuples (as was described in previous review), but now I changed example callable option method to a proper string. Could you somehow specify what exact tuples should be there and how it should look like?